### PR TITLE
Detect flow pkts 6164 v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6164

Describe changes:
- detect: adds flow.pkts_toclient keyword

https://github.com/OISF/suricata-verify/pull/1267
```
SV_BRANCH=pr/1267
```

#9051 with review taken into account

### CODE REVIEW

File: src/detect-flow-pkts.c [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+    sigmatch_table[DETECT_FLOW_PKTS_TO_SERVER].SetupPrefilter = PrefilterSetupFlowPktsToServer;
+}
+
+static int DetectFlowBytesToClientMatch(
+        DetectEngineThreadCtx *det_ctx, Packet *p, const Signature *s, const SigMatchCtx *ctx)
+{
+    if (p->flow == NULL) {
+        return 0;
+    }
+    uint32_t nb = p->flow->tosrcbytecnt;
```
Comment (victorjulien) [inline src/detect-flow-pkts.c:188]: `Flow::tosrcbytecnt` is u64

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/detect-flow-pkts.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-flow-pkts.c:188]: And CI is green :-/

Comment (catenacyber) [reply src/detect-flow-pkts.c:188]: Argh, that looks like a bug in clang `Wimplicit-int-conversion`

If I change u32 to u16, I get as expected 
```
detect-flow-pkts.c:268:28: warning: implicit conversion loses integer precision: 'uint64_t' (aka 'unsigned long long') to 'uint16_t' (aka 'unsigned short') [-Wimplicit-int-conversion]
    uint16_t nb = p->flow->todstbytecnt;
```
But no warning with downcast from u64 to u32...

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/9095

**Pull request closed without merging**